pybamm/solvers/base_solver.py

```diff
@@ -16,6 +16,13 @@
 from pybamm.expression_tree.binary_operators import _Heaviside
 
 
+def validate_max_step(max_step):
```

Comment:
we don't validate other arguments such as tolerances. my guess is that all the solvers will have their own check to make sure max step makes sense, so maybe we should remove this?